app/assets/javascripts/shared/confirm_nav.js.coffee

```diff
@@ -0,0 +1,37 @@
+setupNavConfirmation = () ->
+  # Initialize a variable that determines whether or not to ask
+  # for confirmation to leave the page, and keep track of the
+  # current description content
+  confirm_nav = false
+  desc = ''
+  if $('.summernote').code()
+    desc = $('.summernote').code().trim()
+
+  # Disable the navigation pop-up if the user is saving or canceling
+  $('#content-save-btn').click ->
+    confirm_nav = false
+    return true
+  $('#content-cancel-btn').click ->
+    confirm_nav = false
+    return true
+
+  # Enable the navigation pop-up if the user is editing the description
+  $('#content-edit-btn').click ->
+    confirm_nav = true
+  $('#add-content-image').click ->
+    confirm_nav = true
+
+  # Pop up a warning if the description has changed
+  $(window).on 'beforeunload', ->
+    desc_new = ''
+    if $('.summernote').code()
+      desc_new = $('.summernote').code().trim()
+
+    if (confirm_nav && (desc_new != desc))
+      return "You have attempted to leave this page. If you have made any " +
```

Comment:
Now that I've run the code, I'm confused as to what this is for.

My page reads:
"This page is asking you to confirm that you want to leave - data you have entered may not be saved."